app/com/arpnetworking/metrics/portal/hosts/impl/ConsulClient.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2016 Inscope Metrics Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.hosts.impl;
+
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.arpnetworking.commons.jackson.databind.ObjectMapperFactory;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import net.sf.oval.constraint.NotEmpty;
+import net.sf.oval.constraint.NotNull;
+import play.libs.ws.WSClient;
+import play.libs.ws.WSResponse;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Client for HTTP Consul API.
+ *
+ * @author Ville Koskela (ville dot koskela at inscopemetrics dot com)
+ */
+public final class ConsulClient {
+
+    /**
+     * Calls the Consul API to get a list of hosts.
+     *
+     * @return A Promise containing a {@code List} of {@link Host}
+     */
+    public CompletionStage<ImmutableList<Host>> getHostList() {
+        return _client
+                .url(_baseUrl + "/v1/catalog/nodes")
+                .get()
+                .thenApply(this::parseWSResponse);
+    }
+
+    private ImmutableList<Host> parseWSResponse(final WSResponse response) {
+        try {
+            if (response.getStatus() / 100 != 2) {
+                throw new IOException("Non-200 response from Consul");
```

Comment:
Also, you throw IOException just to catch it and throw it again as a runtime exception. I do not like that at all.